docs/speech-usage.rst

```diff
@@ -145,5 +141,77 @@ words to the vocabulary of the recognizer.
     transcript: Hello, this is a test
     confidence: 0.81
 
+
+Streaming Recognition
+---------------------
+
+The :meth:`~google.cloud.speech.Client.stream_recognize` method converts speech
+data to possible text alternatives on the fly.
+
+.. note::
+    Streaming recognition requests are limited to 1 minute of audio.
+
+    See: https://cloud.google.com/speech/limits#content
+
+.. code-block:: python
+
+    >>> import io
+    >>> from google.cloud import speech
+    >>> client = speech.Client()
+    >>> with io.open('./hello.wav', 'rb') as stream:
+    ...     sample = client.sample(stream=stream, encoding=speech.Encoding.LINEAR16,
+    ...                            sample_rate=16000)
+    ...     for response in client.stream_recognize(sample):
+    ...         print(response.transcript)
+    hello
```

Comment:
So actually, unless you set `interim_results=True`, there will only be one response. Do you think it makes sense in this case to return just one result unless `interim_results=True`?

Seems weird to call `stream_recognize()` and have the returned value be an object in one case and a list of objects in another case. Maybe that's just me though?